packages/web-components/src/components/va-search-input/va-search-input.tsx

```diff
@@ -58,6 +60,15 @@ export class VaSearchInput {
     this.updateSuggestions(this.suggestions);
   }
 
+  /**
+   * Fixes issue where submit event dispatches the initial value of value
+   * instead of the current value of the input field.
+   */
+  @Watch('value')
+  watchValueHandler() {
+    this.value = this.inputRef.value;
```

Comment:
How is this different than the `this.value` assignment in the `handleInput` function?

Reply:
This is a very hacky solution that reconciles `this.value` with what's actually present in the input field. When this component re-renders, `this.value` resets to the initial value of `value`. You can observe this in the DOM by inspecting the element, typing "for" and wait for the component to receive a list of new suggestions. When the component receives the new suggestions, `event.target.value` is reset. This has been the only way that I've been able to make sure that `event.target.value` is correct for the `submit` event. I believe this is happening because we're using `value` like a state variable when it's really a prop, but I haven't found any documentation to really explain it. The assignment in `handleInput` happens before it calls `updateSuggestions` that triggers a re-render when `formattedSuggestions` are updated.